sonarcloud/resource_project_main_branch.go

```diff
@@ -0,0 +1,189 @@
+package sonarcloud
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/reinoudk/go-sonarcloud/sonarcloud/project_branches"
+)
+
+type resourceProjectMainBranchType struct{}
+
+func (r resourceProjectMainBranchType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `This resource manages a project main branch.
+
+Note that certain operations, such as the deletion of a project's main branch configuration, may
+not be permitted by the SonarCloud web API, or may require admin permissions.
+		`,
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.StringType,
+				Computed: true,
+			},
+			"name": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The name of the project main branch.",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.RequiresReplace(),
```

Comment:
Semantically I think it makes more sense to use `tfsdk.RequiresReplace()` on the `project_key`. Changing the name of the project main branch is an update, while changing the `project_key` would mean we actually start managing the attribute of a different resource. 

Practically there is not much of a difference, as the NOOP delete + create is the same as just an update.